### ISSUE TITLE
this.declaration.assignExpression is not a function

### ISSUE THREAD

**Issue by negberts** [rollup/rollup#1587]
We get the following error during angular-seed build with new version 0.49

TypeError: this.declaration.assignExpression is not a function
    at Node.assignExpression (C:\Git\PIM\Pim.Solution\Pim.PortalApp\node_modules\rollup\dist\rollup.js:6781:21)
    at Node.bind (C:\Git\PIM\Pim.Solution\Pim.PortalApp\node_modules\rollup\dist\rollup.js:5971:13)
    at eachChild.child (C:\Git\PIM\Pim.Solution\Pim.PortalApp\node_modules\rollup\dist\rollup.js:5670:34)
    at keys.forEach.key (C:\Git\PIM\Pim.Solution\Pim.PortalApp\node_modules\rollup\dist\rollup.js:5681:5)
    at Array.forEach (native)
    at Node.eachChild (C:\Git\PIM\Pim.Solution\Pim.PortalApp\node_modules\rollup\dist\rollup.js:5674:13)
    at Node.bind (C:\Git\PIM\Pim.Solution\Pim.PortalApp\node_modules\rollup\dist\rollup.js:5670:8)
    at body.forEach.node (C:\Git\PIM\Pim.Solution\Pim.PortalApp\node_modules\rollup\dist\rollup.js:6038:35)
    at Array.forEach (native)
    at Node.bind (C:\Git\PIM\Pim.Solution\Pim.PortalApp\node_modules\rollup\dist\rollup.js:6038:13)

Our gulp task is:

```
import Config from '../../config';
import { writeFile } from 'fs';
import { join } from 'path';

const globals = require('rollup-plugin-node-globals');
const builtins = require('rollup-plugin-node-builtins');

const nodeResolve = require('rollup-plugin-node-resolve');
const commonjs = require('rollup-plugin-commonjs');
const includePaths = require('rollup-plugin-includepaths');
const rollup = require('rollup');

const config = {
    input: join(Config.TMP_DIR, Config.BOOTSTRAP_FACTORY_PROD_MODULE),
    onwarn: function (warning: any) {
        // Suppress this error message... there are hundreds of them. Angular team says to ignore it.
        // https://github.com/rollup/rollup/wiki/Troubleshooting#this-is-undefined
        if (warning.code === 'THIS_IS_UNDEFINED' || warning.code === 'NAMESPACE_CONFLICT') {
            return;
        }

        console.error(warning.code + ': ' + warning.message);
    },
    sourcemap: true,
    treeshake: true,
    name: 'main',
    plugins: [
        includePaths({
            include: {},
            paths: [join(Config.TMP_DIR, 'app')],
            external: [],
            extensions: ['.js', '.json', '.html', '.ts']
        }),
        nodeResolve({
            jsnext: true,
            main: true,
            module: true,
            preferBuiltins: true
        }),
        commonjs({ //See project.config.ts to extend
            include: Config.ROLLUP_INCLUDE_DIR,
            namedExports: Config.getRollupNamedExports()
        }),
        globals(),
        builtins()
    ]
};

export = (done: any) => {
    rollup.rollup(config)
        .then((bundle: any) => {
            return bundle.generate({
                format: 'iife'
            }).then((result: any) => {
                const path = join(Config.TMP_DIR, 'bundle.js');
                writeFile(path,
                    result.code,
                    (error: any) => {
                        if (error) {
                            console.error(error);
                            process.exit(0);
                        }
                        done();
                    });
            });
        })
        .catch((error: any) => {
            console.error(error);
            process.exit(0);
        });
};

```

Is this a rollup error or some error in our own code?

**Comment by lukastaegert** [comment]
It is probably a problem with some refactorings I did recently. I will take a look what might cause this.

**Comment by lukastaegert** [comment]
Thanks for the call, it seems that assigning values to global variables was not handled properly. This should be fixed by #1587. If you still have issues once this one is merged, please file another issue. Until then I guess you should stick with the previous version of Rollup.

**Comment by nuintun** [comment]
The same error.

**Comment by lukastaegert** [comment]
Did you directly use the code from the pull-request? To do this, you need to add something like
"rollup":  "rollup/rollup#pull/1586/head"
to your (dev-)dependencies in your package.json because, as I said, this has not yet been merged.

If the issue is still present, it would be very helpful if you could pinpoint at least roughly which part of your code may cause this. The error message implies that there is some assignment (i.e. something with an "=") that causes this.

**Comment by negberts** [comment]
My error is fixed in 0.49.1
